src/main/java/io/prometheus/cloudwatch/WebServer.java

```diff
@@ -9,40 +14,80 @@
 
 public class WebServer {
+    private static final Logger LOGGER = Logger.getLogger(WebServer.class.getPackage().getName());
 
     public static String configFilePath;
+    public static int port;
 
-    public static void main(String[] args) throws Exception {
-        if (args.length < 2) {
-            System.err.println("Usage: WebServer <port> <yml configuration file>");
-            System.exit(1);
+    private static void enableDebugLogs() {
+      LOGGER.setLevel(Level.FINEST);
+      ConsoleHandler handler = new ConsoleHandler();
+      LOGGER.addHandler(handler);
+      for (Handler h: LOGGER.getHandlers()) {
+        h.setLevel(Level.FINEST);
+      }
+      LOGGER.log(Level.FINEST, "debug logging is on");
+    }
+
+    private static void printUsage() {
+        System.out.println("cloudwatch_exporter [-d] <port> <yml configuration file>");
+        System.out.println("-d,--debug     enable debug logs");
+    }
+
+    private static void parseOptions(String[] args) {
+        // usage can be either:
+        // -d <port> <yml configuration file>
+        // OR
+        // <port> <yml configuration file>
+        int argsOffset = 0;
+        if (args[0].equals("-d")) {
+          enableDebugLogs();
+          argsOffset++;
+        }
+        if ((args.length - argsOffset) < 2) {
+          printUsage();
+          System.exit(0);
         }
+        port = Integer.parseInt(args[argsOffset + 0]);
+        configFilePath = args[argsOffset + 1];
+    }
+
+    public static void main(String[] args) throws Exception {
+        parseOptions(args);
 
-        configFilePath = args[1];
         CloudWatchCollector collector = null;
         FileReader reader = null;
-
         try {
           reader = new FileReader(configFilePath);
           collector = new CloudWatchCollector(new FileReader(configFilePath)).register();
-        } finally {
           reader.close();
+          ReloadSignalHandler.start(collector);
+        } catch (Exception e) {
+          LOGGER.log(Level.SEVERE, "failed to start collector");
+          LOGGER.log(Level.SEVERE, e.toString());
+          System.exit(1);
+        }
+
+        Server server = null;
+        try {
+          server = new Server(port);
+          ServletContextHandler context = new ServletContextHandler();
+          context.setContextPath("/");
+          server.setHandler(context);
+          context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
+          context.addServlet(new ServletHolder(new DynamicReloadServlet(collector)), "/-/reload");
+          context.addServlet(new ServletHolder(new HealthServlet()), "/-/healthy");
+          context.addServlet(new ServletHolder(new HealthServlet()), "/-/ready");
+          context.addServlet(new ServletHolder(new HomePageServlet()), "/");
+          server.start();
+        } catch (Exception e) {
+          LOGGER.log(Level.SEVERE, String.format("failed to start sever on port %d", port));
```

Comment:
server not sever

Reply:
> server not sever

No, the log level is called "SEVERE"

Reply:
I'm not referring to SEVERE, but sever: "failed to start sever on port %d"

Reply:
Haha, sorry, my bad.

Reply:
I made a change suggestion on the typo.